src/Strings.f90

```diff
@@ -113,17 +113,67 @@ MODULE Strings
 
   !> Derived type for an arbitrary length string
   TYPE :: StringType
-    !> The size of the string
-    !>
-    !> This is needed because SIZE(%s) does not reliably return 0 when
-    !> s has been allocated and then unallocated.
-    INTEGER(SIK) :: n=0
-    !> The trim length of the string
-    !>
-    !> This is needed for the TRIM routine
-    INTEGER(SIK) :: ntrim=0
-    !> The string stored as an array of length 1 character strings
-    CHARACTER(LEN=1),ALLOCATABLE :: s(:)
+    !> The stored intrinsic character string
+    CHARACTER(LEN=:),ALLOCATABLE,PRIVATE :: s
+    CONTAINS
+      !> copybrief StringType::toupper_string
+      !> copydetails StringType::toupper_string
+      PROCEDURE,PASS :: upper => toupper_string
+      !> copybrief StringType::toLower_string
+      !> copydetails StringType::toLower_string
+      PROCEDURE,PASS :: lower => toLower_string
+      !> copybrief StringType::at_string
+      !> copydetails StringType::at_string
+      PROCEDURE,PASS :: at_string
+      !> copybrief StringType::at_string_slice
+      !> copydetails StringType::at_string_slice
+      PROCEDURE,PASS :: at_string_slice
+      GENERIC :: at => at_string, at_string_slice
```

Comment:
how did we decide on at instead of substring?

Reply:
I guess I feel like "at" is not descriptive enough

Reply:
So I just extended the single character at to use for a slice from the string, but `at_string_slice` could (maybe should) be `substr`.

Reply:
Implemented `at_string_slice` as `substr`